getUserFeatureFlags/main.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/Real-Dev-Squad/feature-flag-backend/database"
+	"github.com/Real-Dev-Squad/feature-flag-backend/models"
+	"github.com/Real-Dev-Squad/feature-flag-backend/utils"
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+)
+
+func processGetById(userId string) ([]models.FeatureFlagUserMapping, error) {
+
+	db := database.CreateDynamoDB()
+
+	input := &dynamodb.QueryInput{
+		TableName:              aws.String(database.GetTableName(utils.FFUM_TABLE_NAME)),
+		KeyConditionExpression: aws.String(fmt.Sprintf("%v = :uid", utils.UserId)),
+		ExpressionAttributeValues: map[string]*dynamodb.AttributeValue{
+			":uid": {
+				S: aws.String(userId),
+			},
+		},
+	}
+
+	result, err := db.Query(input)
+	if err != nil {
+		log.Println(err)
+		return nil, err
+	}
+	if len(result.Items) == 0 {
+		return nil, nil
+	}
+
+	var listOfFeatureFlagUserMapping []models.FeatureFlagUserMapping
+
+	err = dynamodbattribute.UnmarshalListOfMaps(result.Items, &listOfFeatureFlagUserMapping)
+	if err != nil {
+		log.Println(err)
+		return nil, err
+	}
+
+	return listOfFeatureFlagUserMapping, nil
+}
+
+func handler(req events.APIGatewayProxyRequest) (events.APIGatewayProxyResponse, error) {
+	userId, found := req.PathParameters["userId"]
+	if !found {
+		log.Panic("User ID not passed")
+		return utils.ClientError(http.StatusBadRequest, "User ID not passed in request")
+	}
+	result, err := processGetById(userId)
+	if err != nil {
+		return utils.ServerError(err)
```

Comment:
it would be better if we could use our custom error message, we do not want to expose some critical errors that might reveal our architecture/business models to clients/users.

Reply:
logged errors are being printed on the terminal, but what do we want to show users would go in response? please look into it.

Reply:
Sure noted, will change the push, this code was written by me.

Reply:
> logged errors are being printed on the terminal, but what do we want to show users would go in response? please look into it.

Please try later, something went wrong @RitikJaiswal75 does this sound good?

Reply:
@RitikJaiswal75 resolved this thing in different PR's commit https://github.com/Real-Dev-Squad/feature-flag-backend/pull/32/commits/639e0ae46f9a2b6411fac49779059d9d7c565aca#diff-bab44cea90688750abba6a9360f9fed8ab5565494c1e305b7d6735531875256bR41

Reply:
> @RitikJaiswal75 resolved this thing in different PR's commit https://github.com/Real-Dev-Squad/feature-flag-backend/commit/639e0ae46f9a2b6411fac49779059d9d7c565aca#diff-bab44cea90688750abba6a9360f9fed8ab5565494c1e305b7d6735531875256bR41

Please create and link an issue ticket for it.

Reply:
Ticket link : #40